Userland/Shell/Builtin.cpp

```diff
@@ -906,6 +906,9 @@ int Shell::builtin_time(int argc, const char** argv)
         for (auto& job : run_commands(commands)) {
             block_on_job(job);
             exit_code = job.exit_code();
+            if (job.signaled()) {
```

Comment:
This will produce a surprising result when the commands e.g. contain a pipe, for instance:
```console
$ foo() { echo hi; sleep 1; echo bye }
$ alias bar='foo | head -n1'
$ time -n4 bar
```

Leaving aside the (now) obvious issue with waiting for all the jobs, the `foo` process will get a SIGPIPE and terminate, which would then make `time` quit - but the pipeline succeeds as `head` exits normally.

Reply:
:thinking: I think we can merge this patch, and fix the first bug - which then auto-fixes the second bug :P